potpie/integrations/tests/test_logging.py

```
@@ -0,0 +1,42 @@
+    logger.propagate = False
+    logger.setLevel(logging.INFO)
+    try:
+        structured = get_logger(logger.name).bind(component="oauth")
+        with log_context(request_id="req-123"):
+            structured.info("connected", extra={"obs_fields": {"provider": "linear"}})
+    finally:
+        logger.handlers[:] = old_handlers
+        logger.propagate = old_propagate
+
+    if len(handler.records) != 1:
+        raise AssertionError(f"expected one log record, got {len(handler.records)}")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)